nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1430 @@
+                raise ValueError("Input {} has incorrect shape ({}) for Scalar axis".format(
+                        check_name, getattr(self, check_name).shape))
+
+    @classmethod
+    def from_mapping(cls, mim):
+        """
+        Creates a new Scalar axis based on a CIfTI2 dataset
+
+        Parameters
+        ----------
+        mim : cifti2.Cifti2MatrixIndicesMap
+
+        Returns
+        -------
+        Scalar
+        """
+        names = [nm.map_name for nm in mim.named_maps]
+        meta = [{} if nm.metadata is None else dict(nm.metadata) for nm in mim.named_maps]
+        return cls(names, meta)
+
+    def to_mapping(self, dim):
+        """
+        Converts the hcp_labels to a MatrixIndicesMap for storage in CIfTI2 format
+
+        Parameters
+        ----------
+        dim : int
+            which dimension of the CIfTI2 vector/matrix is described by this dataset (zero-based)
+
+        Returns
+        -------
+        cifti2.Cifti2MatrixIndicesMap
+        """
+        mim = cifti2.Cifti2MatrixIndicesMap([dim], 'CIFTI_INDEX_TYPE_SCALARS')
+        for name, meta in zip(self.name, self.meta):
+            meta = None if len(meta) == 0 else meta
+            named_map = cifti2.Cifti2NamedMap(name, cifti2.Cifti2MetaData(meta))
+            mim.append(named_map)
+        return mim
+
+    def __len__(self):
+        return self.name.size
+
+    def __eq__(self, other):
+        """
+        Compares two Scalars
+
+        Parameters
+        ----------
+        other : Scalar
+            scalar axis to be compared
+
+        Returns
+        -------
+        bool : False if type, length or content do not match
+        """
+        if not isinstance(other, Scalar) or self.size != other.size:
+            return False
+        return np.array_equal(self.name, other.name) and np.array_equal(self.meta, other.meta)
+
+    def __add__(self, other):
+        """
+        Concatenates two Scalars
+
+        Parameters
+        ----------
+        other : Scalar
+            scalar axis to be appended to the current one
+
+        Returns
+        -------
+        Scalar
+        """
+        if not isinstance(other, Scalar):
+            return NotImplemented
+        return Scalar(
+                np.append(self.name, other.name),
+                np.append(self.meta, other.meta),
+        )
+
+    def __getitem__(self, item):
+        if isinstance(item, integer_types):
+            return self.get_element(item)
+        return self.__class__(self.name[item], self.meta[item])
+
+    def get_element(self, index):
+        """
+        Describes a single element from the axis
+
+        Parameters
+        ----------
+        index : int
+            Indexes the row/column of interest
+
+        Returns
+        -------
+        tuple with 2 elements
+        - unicode name of the row/column
+        - dictionary with the element metadata
+        """
+        return self.name[index], self.meta[index]
+
+
+class Label(Axis):
```

Comment:
`LabelAxis` might be a slightly more clear name. `Label` might be confusing even if it's within a `cifti2_axis` module. What do you think @effigies ?

Reply:
I would either add Axis to all of the names (`Scalar` could be unclear as well) or to none for consistency, however as we are always within the `cifti2` environment I think it should be clear enough. @effigies, more thoughts on this would be welcome.

Reply:
I agree that it’s all or none. I think that often the module naming gets a bit lost  in python’s flexible import system. In nibabel we have naming like `Nifti2Image` when it could just be `Image` within a module `nitfi2` so this issue is considered elsewhere. I agree that the last call should come from @effigies

Reply:
Setting aside the question of interpretability when imported as a bare name, which would you find a better description of the object in question: `cifti2.Label` or `cifti2.LabelAxis` (and so on for each `Axis` subclass)? We should consider (our estimate of) the intuition of someone roughly familiar with the CIFTI-2 format but not our data structures. Note we also have `cifti2.Cifti2Label` objects for the XML nodes.

From my perspective as one who's vaguely acquainted with CIFTI-2, and more via code reviews than actual manipulation of files, I'm a little inclined toward `(BrainModel|Parcels|Scalar|Label|Series)Axis`, but I don't feel very strongly about it.

You asked for a last call, so I'd say let's go with @demianw, but if you feel strongly the other way, @MichielCottaar, please feel free to argue further.

I'm going to go through hopefully a last round of review now.